source/geod24/concurrency.d

```diff
@@ -985,6 +986,161 @@ protected:
         bool notified;
     }
 
+    class Channel (T)
+    {
+        this(ulong size = 0)
+        {
+            this.size = size;
+            m_lock = new Mutex;
+        }
+
+        bool write(T val)
+        {
+            bool notify_peer, fiber_was_blocked = false;
+            ChannelQueueEntry qEnt;
+
+            synchronized (m_lock)
+            {
+                notify_peer = m_rxq.length > 0;
+
+                if (m_closed)
+                {
+                    return false;
+                }
+                else if (m_rxq.length > 0)
+                {
+                    qEnt = m_rxq[0];
+                    m_rxq = m_rxq.remove(0);
+
+                    *(qEnt.pVal) = val;
+                }
+                else if (m_buffer.length < size)
```

Comment:
This seems to have a nice property of also handling the case where `size` is zero, but it's much better to be explicit about the intention.

```suggestion
                else if (this.size > 0 && m_buffer.length < this.size)
```

Reply:
And again I suggest renaming it. To something like `size_limit` or `max_size`, etc.

Reply:
I also suggest to add small comments, something like:

```
if (...)  // buffered
{
}
else if (...)  // unbuffered
{
}
```